MediaBrowser.Api/Playback/Hls/DynamicHlsService.cs

```diff
@@ -722,22 +722,155 @@ private bool EnableAdaptiveBitrateStreaming(StreamState state, bool isLiveStream
+            else if (string.Equals(state.ActualOutputVideoCodec, "h265", StringComparison.InvariantCultureIgnoreCase)
+                    || string.Equals(state.ActualOutputVideoCodec.ToLower() == "hevc", StringComparison.InvariantCultureIgnoreCase))
+            {
+                string profile = state.GetRequestedProfiles("h265").FirstOrDefault();
+                return HLSCodecStringFactory.GetH265String(profile, level);
+            }
+            return string.Empty;
+        }
+
+        private void AppendPlaylistCodecsField (StringBuilder field, StreamState state)
+        {
+            // Video
+            string videoCodecs = string.Empty;
+            if (!string.IsNullOrEmpty(state.ActualOutputVideoCodec))
+            {
+                videoCodecs = GetPlaylistVideoCodecs(state);
+            }
+
+            // Audio
+            string audioCodecs = string.Empty;
+            if (!string.IsNullOrEmpty(state.ActualOutputAudioCodec))
+            {
+                audioCodecs = GetPlaylistAudioCodecs(state);
+            }
+
+            if (!string.IsNullOrEmpty(videoCodecs) || !string.IsNullOrEmpty(audioCodecs))
+            {
+                field.Append(",CODECS=\"");
+
+                if (!string.IsNullOrEmpty(videoCodecs) && !string.IsNullOrEmpty(audioCodecs))
+                {
+                    // If both video and audio exists
+                    field.Append(videoCodecs + "," + audioCodecs);
+                }
+                else if (!string.IsNullOrEmpty(videoCodecs))
+                {
+                    // If only video exists
+                    field.Append(videoCodecs);
+                }
+                else if (!string.IsNullOrEmpty(audioCodecs))
+                {
+                    // If only audio exists
+                    field.Append(audioCodecs);
+                }
 
+                field.Append("\"");
+            }
+        }
+
+        private void AppendPlaylistFramerateField(StringBuilder field, StreamState state)
+        {
+            double? framerate = null;
+            if (state.TargetFramerate.HasValue)
+            {
+                framerate = Math.Round(state.TargetFramerate.GetValueOrDefault(), 3);
+
+            }
+            else if (state.VideoStream.RealFrameRate.HasValue)
+            {
+                framerate = Math.Round(state.VideoStream.RealFrameRate.GetValueOrDefault(), 3);
+            }
+
+            if (framerate.HasValue)
+            {
+                field.Append(",FRAME-RATE=\"");
+                field.Append(framerate.Value);
+                field.Append("\"");
+            }
+        }
+
+        private void AppendPlaylistResolutionField(StringBuilder field, StreamState state)
+        {
+            if (state.OutputWidth.HasValue && state.OutputHeight.HasValue)
+            {
+                field.Append(",RESOLUTION=\"");
+                field.Append(state.OutputWidth.GetValueOrDefault());
+                field.Append("x");
```

Comment:
```suggestion
                field.Append('x');
```
Append single characters as `char`s